src/torznab.ts

```diff
@@ -348,80 +348,101 @@ export function indexerDoesSupportMediaType(
 
 export async function* rssPager(
 	indexer: Indexer,
-	pageBackUntil: number,
+	timeSinceLastRun: number,
 ): AsyncGenerator<Candidate, void, undefined> {
-	let earliestSeen = Infinity;
 	const limit = indexer.limits.max;
-	for (let i = 0; i < 10; i++) {
+	const lastSeenGuid: string | undefined = (
+		await db("rss")
+			.where({ indexer_id: indexer.id })
+			.select("last_seen")
+			.first()
+	)?.last_seen;
+	let newLastSeenGuid: string | undefined = lastSeenGuid;
+	let pageBackUntil = 0;
+	const maxPage = 10;
+	let i = -1;
+	while (++i < maxPage) {
 		let currentPageCandidates: Candidate[];
 
 		try {
-			currentPageCandidates = await makeRequest({
-				indexerId: indexer.id,
-				baseUrl: indexer.url,
-				apikey: indexer.apikey,
-				query: { t: "search", q: "", limit, offset: i * limit },
-			});
+			currentPageCandidates = (
+				await makeRequest({
+					indexerId: indexer.id,
+					baseUrl: indexer.url,
+					apikey: indexer.apikey,
+					query: { t: "search", q: "", limit, offset: i * limit },
+				})
+			).sort((a, b) => b.pubDate - a.pubDate);
```

Comment:
could be a good application of `comparing`

Reply:
I think it's mainly beneficial when doing multiple sorts, I find this more straightfoward.